core/tauri/src/app.rs

```diff
@@ -772,12 +772,16 @@ macro_rules! shared_app_impl {
         #[cfg(all(desktop, feature = "tray-icon"))]
         self.manager.tray.icons.lock().unwrap().clear();
         self.manager.resources_table().clear();
-        for (_, window) in self.manager.windows().iter() {
+        for (_, window) in self.manager.windows() {
           window.resources_table().clear();
         }
-        for (_, webview) in self.manager.webviews().iter() {
+        for (_, webview) in self.manager.webviews() {
           webview.resources_table().clear();
         }
+        #[cfg(windows)]
+        for (_, window) in self.manager.windows() {
+          let _ = window.hide();
```

Comment:
how will this effect plugins like `window-state`?

Reply:
This should be run after all other exit event handlers so should be fine

> ```rust
> let event = on_event_loop_event(&app_handle, RuntimeRunEvent::Exit, &manager);
> callback(&app_handle, event);
> app_handle.cleanup_before_exit();
> ```